src/engine/Scene.hpp

```diff
@@ -22,12 +22,26 @@ class Scene : public renderer::IDrawable {
 
     std::multiset<std::shared_ptr<GameObject>, GameObjectComparator> objects;
 
+    int gridLenght;
+    int gridHeight;
+    bool dupsAllowed;
+
 public:
     Scene();
     ~Scene();
 
     void render() const;
     void addObject(std::shared_ptr<GameObject> object);
+    void removeOneObject(std::shared_ptr<GameObject> object);
+    void removeAllObject(std::shared_ptr<GameObject> object);
+    bool existObject(std::shared_ptr<GameObject> object);
+
+    int sizeObjects();
```

Comment:
I feel this could be renamed to `numObjectsInScene` or something similar. Size means something different to me.